chore(data-warehouse): Updated messaging around incremental syncing (#8927)
PostHog/posthog.com: contents/docs/data-warehouse/setup/index.mdx

## Incremental vs full table

On some sources and tables, you can choose the sync method. The options are incremental append replication or full table replication. 

### Incremental append

> With incremental append replication, you only sync new data. This reduces the total number of rows synced and how long it takes to sync. 

Reviewer: Maybe add something highlighting where this is useful. `This is useful for data that is immutable such as invoices` or something like that

> When choosing incremental append replication, you must select a field to identify new data. This is often something like a `created_at` timestamp, or an autoincrementing ID. Not all fields are suitable to be used to identify new and updated data, and so we only support the following types as replication keys:

Reviewer 2: ```suggestion
When choosing incremental append replication, you must select a field to identify new data. This is often something like a `created_at` timestamp, or an autoincrementing ID. Not all fields are suitable to be used to identify new data, and so we only support the following types as replication keys:
```

- `integer` (including `bigint` and `smallint`)
- `datetime`
- `date`
- `timestamp`
- `numeric` (for Snowflake)

> Duplicate data can exist if you use a replication field that gets updated when the row gets updated, such as an `updated_at` field. Updated data will be appened to the table as if it's a new row and any deduplication logic will have to be written in HogQL.

Reviewer 2: ```suggestion
Duplicate data can exist if you use a replication field that gets updated when the row gets updated, such as an `updated_at` field. Updated data will be appened to the table as if it's a new row and any deduplication logic would need to be written in HogQL when you query the data.
```

> Another downside to incremental append syncing is that deletions of data won't be synced to your PostHog data warehouse. You need to use full table refreshes for this.

Reviewer 2: ```suggestion
Another downside to incremental append syncing is that data deletions won't be synced to your PostHog data warehouse. You need to use full table refreshes for this.
```
